tests/integration/test_codecov_workflow_removal.py

```
@@ -0,0 +1,104 @@
+"""
+Comprehensive tests for codecov workflow removal validation.
+
+Tests validate that the codecov.yaml workflow was properly removed and that
+coverage functionality is still available through other means.
+"""
+
+import pytest
+from pathlib import Path
+from typing import List
```

Ruff (0.14.6)

`typing.List` is deprecated, use `list` instead

(UP035)